test/unit/mongo_client.test.js

```diff
@@ -976,35 +1019,121 @@ describe('MongoOptions', function () {
+          beforeEach(async function () {
+            process.env['MONGODB_LOG_ALL'] = 'imFakeToo';
+          });
+          afterEach(async function () {
+            process.env['MONGODB_LOG_ALL'] = 'undefined';
+          });
+
+          context('when client option is not provided', function () {
+            it(`should not throw error, and set component severity to 'off'`, function () {
+              expect(
+                () =>
+                  new MongoClient('mongodb://a/', {
+                    [loggerFeatureFlag]: true,
+                    mongodbLogComponentSeverities: {}
+                  })
+              ).to.not.throw(MongoAPIError);
+              const client = new MongoClient('mongodb://a/', {
                 [loggerFeatureFlag]: true,
-                mongodbLogMaxDocumentLength: -290
-              })
-          ).to.throw(MongoParseError);
+                mongodbLogComponentSeverities: {}
+              });
+              expect(client.mongoLogger.componentSeverities.default).to.equal('off');
+            });
+          });
+          context('when valid client option is provided', function () {
+            it('should not throw error, and set component severity to client value', function () {
+              expect(
+                () =>
+                  new MongoClient('mongodb://a/', {
+                    [loggerFeatureFlag]: true,
+                    mongodbLogComponentSeverities: { default: 'emergency' }
+                  })
+              ).to.not.throw(MongoAPIError);
+              const client = new MongoClient('mongodb://a/', {
+                [loggerFeatureFlag]: true,
+                mongodbLogComponentSeverities: { default: 'emergency' }
+              });
+              expect(client.mongoLogger.componentSeverities.default).to.equal('emergency');
+            });
+          });
+        });
+      });
+    });
+    describe('mongodbLogMaxDocumentLength', function () {
+      context('when mongodbLogMaxDocumentLength is in options', function () {
+        context('when env option for MONGODB_LOG_MAX_DOCUMENT_LENGTH is not provided', function () {
+          it('it stores value for maxDocumentLength correctly', function () {
+            const client = new MongoClient('mongodb://a/', {
+              [loggerFeatureFlag]: true,
+              mongodbLogMaxDocumentLength: 290
+            });
+            expect(client.options.mongoLoggerOptions.maxDocumentLength).to.equal(290);
+          });
+          it('it throws error for negative input', function () {
+            expect(
+              () =>
+                new MongoClient('mongodb://a/', {
+                  [loggerFeatureFlag]: true,
+                  mongodbLogMaxDocumentLength: -290
+                })
+            ).to.throw(MongoParseError);
+          });
+        });
+        context('when env option for MONGODB_LOG_MAX_DOCUMENT_LENGTH is provided', function () {
```

Comment:
Should we add test cases when MONGODB_LOG_MAX_DOCUMENT_LENGTH is provided and mongodbLogMaxDocumentLength is negative, and also when MONGODB_LOG_MAX_DOCUMENT_LENGTH is provided but negative?

Reply:
Good idea, just added them in